spec/controllers/settings/dex_connector_oidcs_controller_spec.rb

```diff
@@ -0,0 +1,371 @@
+    end
+
+    context "with rendered views" do
+      render_views
+
+      it "shows an error message for empty name" do
+        VCR.use_cassette("oidc/validate_connector", allow_playback_repeats: true, record: :none) do
+          get :create, validate: false, dex_connector_oidc: {
+            name:          "",
+            provider_url:  good_provider,
+            callback_url:  "http://well.formed.but.invalid/", # can't be blank in form
+            basic_auth:    true, # also can't be blank - only true/false
+            client_id:     "client",
+            client_secret: "secret"
+          }
+        end
+        expect(response).to render_template("new")
+        expect(response.body).to match(/Name can('|&#39;)t be blank/)
+        # TODO: expect save button to be disabled
+      end
+
+      it "shows an error message for empty provider" do
+        get :create, validate: false, dex_connector_oidc: {
+          name:          "good oidc",
+          provider_url:  "",
+          callback_url:  "http://well.formed.but.invalid/", # can't be blank in form
+          basic_auth:    true, # also can't be blank - only true/false
+          client_id:     "client",
+          client_secret: "secret"
+        }
+        expect(response).to render_template("new")
+        expect(response.body).to match(
+          /Provider Url (can('|&#39;)t be blank|is not a valid HTTP URL)/i
+        )
+        # TODO: expect save button to be disabled
+      end
+
+      it "shows an error message for empty client id" do
+        VCR.use_cassette("oidc/validate_connector", allow_playback_repeats: true, record: :none) do
+          get :create, validate: false, dex_connector_oidc: {
+            name:          "good oidc",
+            provider_url:  good_provider,
+            callback_url:  "http://well.formed.but.invalid/", # can't be blank in form
+            basic_auth:    true, # also can't be blank - only true/false
+            client_id:     "",
+            client_secret: "secret"
+          }
+        end
+        expect(response).to render_template("new")
+        expect(response.body).to match(/Client (Id )?can('|&#39;)t be blank/i)
+        # TODO: expect save button to be disabled
+      end
+
+      it "shows an error message for empty client secret" do
+        VCR.use_cassette("oidc/validate_connector", allow_playback_repeats: true, record: :none) do
+          get :create, validate: false, dex_connector_oidc: {
+            name:          "good oidc",
+            provider_url:  good_provider,
+            callback_url:  "http://well.formed.but.invalid/", # can't be blank in form
+            basic_auth:    true, # also can't be blank - only true/false
+            client_id:     "client",
+            client_secret: ""
+          }
+        end
+        expect(response).to render_template("new")
+        expect(response.body).to match(/Client Secret can('|&#39;)t be blank/i)
+        # TODO: expect save button to be disabled
```

Comment:
TODO's should be resolved before merge / accept of the PR.

Reply:
Fixed in 86830cb